### ISSUE TITLE
pywbem 1.0.0 beta release

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#2255]
Because pywbem 1.0.0 has incompatible changes, and because pywbemtools needs to have a chance to test against it in its CI setup, it is suggested to release a beta version of pywbem 1.0.0 on Pypi.

This issue is a release planning issue that captures what needs to be done to be ready for such a release.

1. Type of release.
    [PEP 440](https://www.python.org/dev/peps/pep-0440/#pre-releases) defines tyes of pre-releases as alpha, beta, or release candidate. Given that we are pretty much done the release we talk about would either be a beta or a release candidate. My perception about a release candidate is that we would think everything is ready, while for a beta, we don't think so. So it seems to me that beta would be appropriate, i.e. version 1.0.0b1

2. Duration of the beta phase.
   We discussed this and think 4 weeks would be appropriate (between 1.0.0b1 release and 1.0.0 final release). During that time, user feedback will be worked in.

3. [List of all open 1.0.0b1 issues](https://github.com/pywbem/pywbem/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0b1)

**Comment by andy-maier** [comment]
It turns out pre-releases are not as easy as I thought. There is some inconsistent behavior of pip that I did not fully grasp at this point.

First, pip intends to follow the latest PEP on versioning, i.e. PEP440: https://www.python.org/dev/peps/pep-0440/#handling-of-pre-releases

However, there are some pip issues that discuss deviations from PEP440 and surprises for users, and so far have either be closed or have not come to a conclusion:
* https://github.com/pypa/pip/issues/4969
* https://github.com/pypa/pip/issues/7579
* https://discuss.python.org/t/version-specifier-sets-and-prereleases/3000

**Comment by andy-maier** [comment]
I did some experiments with the "apispec" package, which has some pre-release versions before its 1.0.0 version:
`..., 0.39.0, 1.0.0b1, 1.0.0b2, 1.0.0b3, 1.0.0b4, 1.0.0b5, 1.0.0b6, 1.0.0rc1, 1.0.0, 1.1.0, ...`

One important thing to understand when doing experiments is that pip accepts already installed versions even if they are pre-releases but pre-releases were excluded in the version specification. That is consistent with PEP 440. So the resulting package with a given version specification may depend on whether it was already installed or not. The following experiments therefore were all done when the package was not installed, i.e. it was uninstalled in between the experiments.

According to the discussion in the issues listed above, another thing to understand is that the presence of pre-releases in the version specification seems to enable the inclusion of pre-releases, even when the `--pre` option was not specified. The `--pre` option alone (i.e. without pre-releases in the version specification) also does that. It is not clear whether the exact rules for these two cases (inclusion of pre-releases triggered by pre-releases in version specification vs. by `--pre` option) are the same or different (and that is partly debated in those issues).

Here are some experiments, and my judgement of the result:
* `pip install 'apispec>0.39.0,<1.0.0'` - does not find a version - OK
* `pip install 'apispec>0.39.0,<1.0.0' --pre` - does not find a version - STRANGE
* `pip install 'apispec>0.39.0,<1.0.0rc1'` - does not find a version - STRANGE
* `pip install 'apispec>0.39.0,<=1.0.0rc1'` - installs `apispec-1.0.0rc1` - OK
* `pip install 'apispec>=1.0.0b1,<1.0.0'` - does not find a version - STRANGE
* `pip install 'apispec>=1.0.0b1,<1.0.0' --pre` - does not find a version - STRANGE

**Comment by andy-maier** [comment]
Now, the pywbem case is somewhat simpler than that, because we will have a first pre-release version of 1.0.0 as the latest version, and the question is just how we would have dependent packages or users specify the installation of it.

I found the "kfp" package that has a pre-release version of its future 1.0.0 as the latest version:
`..., 0.5.1, 1.0.0rc1`

Note that this package has a lot of dependencies on top of what pywbem and pywbemtools need, so it is best to use a separate virtualenv for any experiments.

Again, I uninstalled it in between the experiments. I used pip 20.1.1 on Python 3.8.

Here are the relevant experiments:
* `pip install 'kfp>=1.0.0rc1'` - installs `kfp-1.0.0rc1` - OK
* `pip install 'kfp>0.5.1'` - does not find a version - OK
* `pip install 'kfp>0.5.1' --pre` - installs `kfp-1.0.0rc1` - OK
